iOS/CourseSceneDelegate.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  Created for xikolo-ios under MIT license.
+//  Copyright © HPI. All rights reserved.
+//
+
+import Common
+import UIKit
+
+@available(iOS 13.0, *)
+class CourseSceneDelegate: UIResponder, UIWindowSceneDelegate {
+
+    private lazy var courseNavigationController: CourseNavigationController = {
+        let courseNavigationController = R.storyboard.course.instantiateInitialViewController().require()
+
+        #if DEBUG
+        if ProcessInfo.processInfo.arguments.contains("-forceDarkMode") {
+            courseNavigationController.overrideUserInterfaceStyle = .dark
+        }
+        #endif
+
+        return courseNavigationController
+    }()
+
+    var window: UIWindow?
+
+    func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
+        if let userActivity = connectionOptions.userActivities.first ?? session.stateRestorationActivity {
+            print(userActivity)
+        }
+
+        let userActivity = connectionOptions.userActivities.first ?? session.stateRestorationActivity
+        guard let courseId = userActivity?.userInfo?["courseID"] as? String else { return } // XXX: really return?
```

Comment:
This there a better way than just returning? If not, the comment can be removed.